### ISSUE TITLE
Is there an error in the interface name " cdef cppclass CFunctionDescriptorInterface  "ray::CFunctionDescriptorInterface "?

### ISSUE THREAD

**Issue by raychen911** [ray-project/ray#41553]
### What happened + What you expected to happen

1、Code in [function_descriptor.pxd](https://github.com/ray-project/ray/blob/master/python/ray/includes/function_descriptor.pxd)

     ` cdef cppclass CFunctionDescriptorInterface \
            "ray::CFunctionDescriptorInterface":`

      There is no class **ray::CFunctionDescriptorInterface** defined in the file [function_descriptor.h](https://github.com/ray-project/ray/blob/master/src/ray/common/function_descriptor.h)
2、Code should be modified to the following form:
     ` cdef cppclass CFunctionDescriptorInterface \
            "ray::FunctionDescriptorInterface":`

### Versions / Dependencies

ray： 2.8.0

### Reproduction script

Just code

### Issue Severity

None

**Comment by iycheng** [comment]
@raychen911  you are right. Do you want to create a PR to fix it? I'll help the review.

**Comment by rkooo567** [comment]
@iycheng is this class even used? Why isn't this caught by any CI?

**Comment by cszhu** [comment]
Hello! This P3 issue has seen no activity in at least a year. It will be closed in 2 weeks as part of ongoing cleanup efforts.

Please remove the `pending-cleanup` label if you believe this issue should remain open.

Thanks for contributing to Ray!
